internal/provider/file/file.go

```diff
@@ -76,9 +76,12 @@ func (p *Provider) Start(ctx context.Context) error {
 		return fmt.Errorf("failed to load resources into store: %w", err)
 	}
 
+	allDirs := path.GetSubDirs(initDirs.UnsortedList())
```

Comment:
can we move this directly into `ListDirsAndFiles`?

Reply:
I have merged it with the `LoadAndStore` function in `resourceStore`. You can't merge it with `ListDirsAndFiles` as that is a function that just loops over the provided list and separates them into two groups based on the output of `isDir.` Since the same does nothing to traverse the directories, it does not make sense to merge it there. Have added it to `traverseDirectory` function in `internal/provide/file/resources.go` instead!